app-media-image-capture.d.ts

```diff
@@ -0,0 +1,158 @@
+ * page for details.
+ *
+ * If you wish to support browsers that do not implement the Image Capture API,
+ * please consider including a polyfill such as
+ * [this one](https://github.com/GoogleChrome/imagecapture-polyfill) in your app.
+ * Note that most polyfills will not enable full functionality, but they should give
+ * your app a reasonable fallback in browsers that do not natively support the API.
+ *
+ * ELEMENT DESIGN NOTE: Many of the properties of this element have a default
+ * configuration value of `null`. While this is weird, it is important to keep the
+ * behavior of multi-property observers in the element implementation normalized
+ * across Polymer 1.x and 2.x. Expect this aspect of the API to change as the
+ * element graduates from the current hybrid implementation to something that can
+ * rely on Polymer >=2.x observer semantics.
+ */
+interface AppMediaImageCaptureElement extends Polymer.Element {
+
+  /**
+   * The input media stream to capture images from. Note that the
+   * stream must have at least one video track in order to be a suitable
+   * source for image capture.
+   */
+  stream: MediaStream|null;
+
+  /**
+   * MediaStreams can have multiple video tracks. This property enables
+   * you to configure the index of the video track to use.
+   */
+  trackIndex: number|null|undefined;
+
+  /**
+   * The video track selected from the input MediaStream. This track will
+   * be the source for any images captured.
+   */
+  readonly videoTrack: MediaStreamTrack|null;
+
+  /**
+   * An ImageCapture instance associated with the selected video track.
+   */
+  readonly imageCapture: ImageCapture|null;
+
+  /**
+   * The last photo captured by the image capture instance.
+   */
+  readonly lastPhoto: Blob|null;
+
+  /**
+   * The last frame grabbed by the image capture instance.
+   */
+  readonly lastFrame: ImageBitmap|null;
+
+  /**
+   * The PhotoCapabilities for the device providing the image data
+   * associated with the chosen video track. This object contains
+   * information about the minimum, maximum, current and incremental
+   * values for various camera settings.
+   */
+  readonly photoCapabilities: object|null|undefined;
+
+  /**
+   * The MediaTrackCapabilities for the device providing the image data
+   * associated with the chosen video track. This object contains
+   * information about the minimum, maximum, current and incremental
+   * values for various camera settings.
+   */
+  readonly trackCapabilities: object|null|undefined;
+
+  /**
+   * The PhotoSettings that will be used to configure the ImageCapture
+   * instance used by this element. This configuration is generated
+   * based on the individually configured properties on this element.
+   * A full list of configurable properties can be found
+   * [here](https://w3c.github.io/mediacapture-image/##photosettings-section).
+   */
+  readonly photoSettings: PhotoSettings;
+  fillLightMode: Polymer.AppMedia.FillLightMode|null;
```

Comment:
Does a type need to be generated for `Polymer.AppMedia.FillLightMode`?

Reply:
@aomarks I added the `@namespace` for `Polymer.AppMedia`, but it does not seem to detect that the other modes are also added to this namespace. Do you maybe know what is going wrong here?